pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix

```diff
@@ -120,27 +120,39 @@ let
       xorg.libXtst
     ];
 
-    postInstall = ''
-      # jni.h expects jni_md.h to be in the header search path.
-      ln -sf $out/include/linux/*_md.h $out/include/
-
-      mkdir -p $out/share
-      # move files in $out like LICENSE.txt
-      find $out/ -maxdepth 1 -type f -exec mv {} $out/share \;
-      # symbolic link to $out/lib/svm/LICENSE_NATIVEIMAGE.txt
-      rm -f $out/LICENSE_NATIVEIMAGE.txt
-
-      # copy-paste openjdk's preFixup
-      # Set JAVA_HOME automatically.
-      mkdir -p $out/nix-support
-      cat > $out/nix-support/setup-hook << EOF
-      if [ -z "\''${JAVA_HOME-}" ]; then export JAVA_HOME=$out; fi
-      EOF
-
-      wrapProgram $out/bin/native-image \
-        --prefix PATH : ${binPath} \
-        ${toString (map (l: "--add-flags '-H:CLibraryPath=${l}/lib'") cLibs)}
-    '';
+    postInstall =
+      let
+        darwinArgs = lib.optionals stdenv.hostPlatform.isDarwin [
```

Comment:
Almost sure that we also want to apply those to non-Darwin systems.

Reply:
Maybe. But for the reasons unknown it works for me on a variety of Linux machines, only Darwin is broken.

Reply:
The reason I am assuming it works is because on Linux, GraalVM doesn't need any extra libraries. So it works, unless the library you're trying to build needs some third party C library (via JNI for example).

Reply:
Ok, I've changed that.